lib/elixir/lib/module/parallel_checker.ex

```diff
@@ -16,15 +16,36 @@ defmodule Module.ParallelChecker do
   end
 
   def verify(modules, schedulers) do
-    {:ok, server} = :gen_server.start_link(__MODULE__, [modules, schedulers], [])
+    modules = prepare_modules(modules)
```

Comment:
Sorry for the nitpick but you want to do this in the server. That's because parsing the binary will emit a ton of AST for runtime modules and those are expensive to copy.

Reply:
So the approach is to collect results until you receive a DOWN message from the parallel checker process.

Reply:
My suggestion is to have this function receive both runtime and compile modules, so we can make the full decision on how to handle each here, so we don't have to change the parallel compiler when making said decisions. Then we will send {module, map_or_binary} to the parallel checker, where a map is sent for compiled and the binary for runtime.